src/Tools/dotnet-monitor/ParameterCapturing/CaptureParametersOperation.cs

```diff
@@ -160,7 +160,14 @@ public async Task ExecuteAsync(Stream outputStream, CancellationToken token)
                     token);
 
                 await _capturingStartedCompletionSource.Task.WaitAsync(token).ConfigureAwait(false);
-                await _capturingStoppedCompletionSource.Task.WaitAsync(token).ConfigureAwait(false);
+
+                /* parametersWriter.WaitAsync() successfully completes only after parametersWriter.DisposeAsync() is called.
+                   It will only complete earlier if there's a fault and in that case we want to catch the exception and stop capturing.
+                */
+                await Task.WhenAny([
```

Comment:
Do we need the `.WaitAsync()` on the tasks here? My understanding is that that we can omit those?

We should also `.Unwrap()` the return value here allow any exceptions from the returning task to properly propagate up (that or capture the task returning from this line and explicitly await it)

e.g. `await Task.WhenAny([.., ...]).Unwrap().ConfigureAwait(false)`